src/components/Modals/FiatRamps/hooks/useFiatRampCurrencyList.ts

```diff
@@ -21,48 +26,75 @@ export const useFiatRampCurrencyList = (fiatRampProvider: FiatRamp) => {
-            supportedFiatRamps[fiatRampProvider].minimumSellThreshold ?? 0,
+      try {
+        if (!wallet) return []
+        return assets
+          .filter(asset => Object.keys(balances).includes(asset.assetId))
+          .map(asset => {
+            const reduxAsset = reduxAssets[asset.assetId]
+            const fiatBalance = bnOrZero(balances?.[asset.assetId]?.fiat)
+            const minimumSellThreshold = bnOrZero(
+              supportedFiatRamps[fiatRampProvider].minimumSellThreshold ?? 0,
+            )
+            return {
+              ...asset,
+              name: reduxAsset.name,
+              symbol: reduxAsset.symbol,
+              disabled: !isAssetSupportedByWallet(asset?.assetId ?? '', wallet),
+              cryptoBalance: bnOrZero(balances?.[asset.assetId]?.crypto),
+              fiatBalance,
+              isBelowSellThreshold: fiatBalance.lt(minimumSellThreshold),
+            }
+          })
+          .sort((a, b) =>
+            a.fiatBalance.gt(0) || b.fiatBalance.gt(0)
+              ? b.fiatBalance.minus(a.fiatBalance).toNumber()
+              : a.name.localeCompare(b.name),
           )
-          return {
-            ...asset,
-            name: reduxAsset.name,
-            symbol: reduxAsset.symbol,
-            disabled: !isAssetSupportedByWallet(asset?.assetId ?? '', wallet),
-            cryptoBalance: bnOrZero(balances?.[asset.assetId]?.crypto),
-            fiatBalance,
-            isBelowSellThreshold: fiatBalance.lt(minimumSellThreshold),
-          }
-        })
-        .sort((a, b) =>
-          a.fiatBalance.gt(0) || b.fiatBalance.gt(0)
-            ? b.fiatBalance.minus(a.fiatBalance).toNumber()
-            : a.name.localeCompare(b.name),
+      } catch (err) {
+        moduleLogger.error(
+          err,
+          { fn: 'addSellPropertiesAndSort' },
+          'An error happened sorting the fiat ramp sell assets',
         )
+        return []
+      }
     },
     [balances, fiatRampProvider, reduxAssets, wallet],
   )
 
   const addBuyPropertiesAndSort = useCallback(
     (assets: FiatRampAsset[]): FiatRampAsset[] => {
       if (!wallet) return []
-      return assets
-        .map(asset => {
-          const reduxAsset = reduxAssets[asset.assetId]
-          return {
-            ...asset,
-            name: reduxAsset.name,
-            symbol: reduxAsset.symbol,
-            disabled: !isAssetSupportedByWallet(asset.assetId, wallet),
-          }
-        })
-        .sort((a, b) => a.name.localeCompare(b.name))
+      try {
+        return assets
+          .map(asset => {
+            const reduxAsset = reduxAssets[asset.assetId]
+            if (!reduxAsset) {
+              return {
+                ...asset,
+                disabled: true,
+              }
+            }
+
+            return {
+              ...asset,
+              name: reduxAsset.name,
+              // Symbol is passed down to the link generation. We can't expect the redux asset
```

Comment:
I moved the mapping into lib and implemented it on coinbase and on the others.

The assetIdTo(Banxa/Gem)AssetId functions were unused in web, as we were just using the symbols on the redux assets (which could be wrong). I changed to pass in the assetId then used the mapping to get the ticker and renamed the functions to assetIdToBanxaTicker, etc. to avoid confusion.